Calling/src/app/App.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GroupCallLocator, GroupLocator, TeamsMeetingLinkLocator } from '@azure/communication-calling';
+import { CommunicationUserIdentifier } from '@azure/communication-common';
+import { DEFAULT_COMPONENT_ICONS } from '@azure/communication-react';
+import { initializeIcons, registerIcons, Spinner } from '@fluentui/react';
+import React, { useEffect, useState } from 'react';
+import {
+  buildTime,
+  callingSDKVersion,
+  createGroupId,
+  fetchTokenResponse,
+  getGroupIdFromUrl,
+  getTeamsLinkFromUrl,
+  isMobileSession,
+  isOnIphoneAndNotSafari,
+  isSmallScreen,
+  navigateToHomePage
+} from './utils/AppUtils';
+import { CallError } from './views/CallError';
+import { CallScreen } from './views/CallScreen';
+import { EndCall } from './views/EndCall';
+import { HomeScreen } from './views/HomeScreen';
+import { UnsupportedBrowserPage } from './views/UnsupportedBrowserPage';
+
+console.log(
+  `ACS sample calling app. Last Updated ${buildTime} Using @azure/communication-calling:${callingSDKVersion}`
+);
+
+initializeIcons();
+registerIcons({ icons: DEFAULT_COMPONENT_ICONS });
+
+type AppPages = 'home' | 'call' | 'endCall';
+
+const webAppTitle = document.title;
+
+const App = (): JSX.Element => {
+  const [page, setPage] = useState<AppPages>('home');
+
+  // User credentials to join a call with - these are retrieved from the server
+  const [token, setToken] = useState<string>();
+  const [userId, setUserId] = useState<CommunicationUserIdentifier>();
+  const [userCredentialFetchError, setUserCredentialFetchError] = useState<boolean>(false);
+
+  // Call details to join a call - these are collected from the user on the home screen
+  const [callLocator, setCallLocator] = useState<GroupLocator | TeamsMeetingLinkLocator>(createGroupId());
+  const [displayName, setDisplayName] = useState<string>('');
+
+  // Get Azure Communications Service token from the server
+  useEffect(() => {
+    (async () => {
+      try {
+        const { token, user } = await fetchTokenResponse();
+        setToken(token);
+        setUserId(user);
+      } catch (e) {
+        console.error(e);
+        setUserCredentialFetchError(true);
+      }
+    })();
+  }, []);
+
+  const supportedBrowser = !isOnIphoneAndNotSafari();
+  if (!supportedBrowser) {
+    return <UnsupportedBrowserPage />;
+  }
+
+  if (isMobileSession() || isSmallScreen()) {
+    console.log('ACS Calling sample: This is experimental behaviour');
```

Comment:
consider removing this console log as the composite supports mobile now

Reply:
in fact we should add a workitem that detects landscape (well smaller than min supported dimensions of composite) and displays something in that case about unsupported

Reply:
Created under work item
https://skype.visualstudio.com/SPOOL/_workitems/edit/2693377